libraries/message/src/lib.rs

```diff
@@ -29,9 +33,13 @@ fn versions_compatible(
     crate_version: &semver::Version,
     specified_version: &semver::Version,
 ) -> Result<bool, String> {
-    let req = semver::VersionReq::parse(&crate_version.to_string()).map_err(|error| {
-        format!("failed to parse crate version `{crate_version}` as `VersionReq`: {error}")
-    })?;
+    let current_version = semver::Version::parse(env!("CARGO_PKG_VERSION"))
+        .map_err(|error| format!("failed to parse current version: {error}"))?;
+    let req = semver::VersionReq::parse(&format!(
+        "~{}.{}.0",
+        current_version.major, current_version.minor
+    ))
+    .map_err(|error| format!("failed to set allowing patch version of messages: {error}"))?;
```

Comment:
Could you clarify why this is needed? What is the issue/error with the current code?

Reply:
Ok, so the current code will fail if we send a patched version message. When we bump dora-message to 4.0.1 it will not pass the semver that is exactly the same version.

With `~4.0.0`, we allow for patch version.

Reply:
The thing is we might want to push updates to messages that are unrelated to the node api. For example adding a new message between the cli and the coordinator or what not, and this will allow us to do this.

Reply:
Sorry, actually, double checking on this at: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c5eaa703cadac747d68985cd10faba7a

It seems the blocking piece is     `let matches = req.matches(&specified_version) || specified_dora_req.matches(crate_version);`

I'm going to replace this piece.

Reply:
> It seems the blocking piece is `let matches = req.matches(&specified_version) || specified_dora_req.matches(crate_version);`

I'm a bit confused what the issue of that piece is. We compare in both directions and return true when either one matches.

Given that we're using semver conventions for the message crate, [caret requirements](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#caret-requirements) seemed like the best match. We can also do tillde requirements if you prefer, this would be a bit stricter. For example, 4.1.0 is semver-compatible with 4.0.0, but not tilde-compatible.

Reply:
Ok, my bad I confused `||` with an `and` sign instead of a `or` sign.

I guess we can leave is as is in that case